### PR TITLE
create action list guidelines

### LINKED ISSUE #2359: Create design guidelines for "Action List" component
Want to create guidelines for this component that has no guidelines yet: https://www.patternfly.org/v4/components/action-list/design-guidelines

### PR DESCRIPTION
**Pull request by mmenestr**
closes #2359 

@mcarrano @afialkoff would love your review on this PR! I've basically just created guidelines for the action list component.

### CODE REVIEW

File: packages/v4/src/content/design-guidelines/components/action-list/action-list.md
Comment (mcarrano) [inline packages/v4/src/content/design-guidelines/components/action-list/action-list.md:8]: "data list" should be plural for consistency with other items.

**Review by mcarrano — CHANGES_REQUESTED** [review]
One super-small grammatical error. Otherwise looks great.

File: packages/v4/src/content/design-guidelines/components/action-list/action-list.md
Comment (mmenestr) [reply packages/v4/src/content/design-guidelines/components/action-list/action-list.md]: @mcarrano fixed!

**Review by mcarrano — DISMISSED** [review]
Looks great @mmenestr !

File: packages/v4/src/content/design-guidelines/components/action-list/action-list.md
Comment (afialkoff) [inline packages/v4/src/content/design-guidelines/components/action-list/action-list.md:8]:
```suggestion
Use an action list to determine which spacing guidelines to use for a group of actions in [toolbars](/components/toolbar), [modals](/components/modal), [forms](/components/form), [data lists](/components/data-list), [wizards](/components/wizard), and more. 

Action lists are especially useful for situations that call for more than standard spacing between elements. For example, buttons are normally separated by 16px, but cases like cancel buttons in wizards call for 48px of spacing from other actions instead. Using an action list upholds spacing guidelines across different use cases.
```

Suggested restructuring for the introduction that combines the first two sentences into one and fleshes out a bit more context for the button use case example. I also rewrote the last sentence of avoid saying "ensure," since we try to avoid absolutes in most RH/PF content!

File: packages/v4/src/content/design-guidelines/components/action-list/action-list.md
Comment (afialkoff) [inline packages/v4/src/content/design-guidelines/components/action-list/action-list.md:21]:
```suggestion
In this example, an action list is used for the actions within a wizard. The action list allows for 16px spacing to be maintained between the main action buttons, and 48px of spacing to be maintained between them and the cancel button.
```

It seems like "action" should be plural here — feel free to reject this suggestion if not! Also rewrote "maintenance" to its verb form, "maintained."

**Review by afialkoff — CHANGES_REQUESTED** [review]
Looks great, @mmenestr! I left just a few small content edits and suggestions. 👍

**Comment by mmenestr** [comment]
@afialkoff @mcarrano this is ready for another look! Addressed all your comments, Alana, thank you!!

**Comment by afialkoff** [comment]
Looks great, @mmenestr! Thanks for your work on these guidelines! 🙌

**Review by mcarrano — APPROVED** [review]
Looks good to me! @afialkoff you need to approve if you are good with this before it can be merged. I will merge once that happens.

**Review by afialkoff — APPROVED** [review]

**Comment by afialkoff** [comment]
@mcarrano just approved, should be all set! :)

**Pull request merged**
